frontend/src/components/main/Main.js

```diff
@@ -87,7 +98,16 @@ export default function Main(props) {
             className="address-input"
             disabled={ props.fetching }
             value={input2}
-            onChange={ (e) => setInput2(e.target.value) }
+            onChange={ (e) => {
+              setInput2(e.target.value)
+              if(vaultStore.balanceOf.value !== 0){
```

Comment:
what if negative or non number? let create an utility function for vailditing a number and checking it with a given min and max?

Reply:
sure